pilot/pkg/serviceregistry/kube/controller/remotekubeconfigcontroller.go

```diff
@@ -0,0 +1,187 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/pkg/log"
+)
+
+const (
+	initialSyncSignal = "INIT"
+	maxRetries        = 5
+)
+
+// RemoteKubeConfigController is the controller implementation for `istio-kubeconfig` secret
+type RemoteKubeConfigController struct {
+	kubeclientset kubernetes.Interface
+	queue         workqueue.RateLimitingInterface
+	informer      cache.SharedIndexInformer
+	mu            sync.RWMutex
+	initialSync   bool
+}
+
+// NewRemoteKubeConfigController returns a pointer to a newly constructed RemoteKubeConfigController instance.
+func NewRemoteKubeConfigController(kubeClient kubernetes.Interface, ns, name string) *RemoteKubeConfigController {
```

Comment:
I think we also want to expand this controller to multicluster secret, right?

Reply:
no, multicluster secret is using its own controller under `pkg/kube/secretcontroller/secretcontroller.go`

Reply:
IC, this is only meant to support the config cluster's kubeconfig. Maybe we could extend it to support multi clusters later